src/Polly.Core/CircuitBreaker/CircuitBreakerStrategyOptions.TResult.cs

```diff
@@ -69,6 +69,14 @@ public class CircuitBreakerStrategyOptions<TResult> : ResilienceStrategyOptions
     public TimeSpan BreakDuration { get; set; } = CircuitBreakerConstants.DefaultBreakDuration;
 #pragma warning restore
 
+    /// <summary>
+    /// Gets or sets the function responsible for dynamically generating the break duration based on the health failure count.
+    /// </summary>
+    /// <value>
+    /// A function that takes an integer representing the health failure count and returns a TimeSpan indicating the break duration.
+    /// </value>
+    public Func<int, TimeSpan>? BreakDurationGenerator { get; set; }
```

Comment:
```suggestion
    public Func<BreakDurationGeneratorArguments, ValueTask<TimeSpan>>? BreakDurationGenerator { get; set; }
```
To align with Polly v8 design:
https://www.pollydocs.org/extensibility/index.html#arguments

Reply:
Thank you for the suggestion. I agree that introducing BreakDurationGeneratorArguments will enhance the extensibility of BreakDurationGenerator. I will make the necessary changes to implement this approach.